### PR TITLE
Update to eslint v9

### PR DESCRIPTION
**Pull request by Nemo157**
Switch to using `deno` as that allows referring to the config packages without having to setup a `package.json` and other stuff required when running `eslint` through `node` now.

### CODE REVIEW

**Comment by Nemo157** [comment]
I have an alternative using `node` with a `package.json` instead if anyone has opinions on which to use: https://github.com/Nemo157/docs.rs/commit/2f41680fd1e02274252ab0450d687e43cbea6dc8

File: eslint.config.js
Comment (Suyashtnt) [inline eslint.config.js:71]: Why does the eslint config have the node globals? it should be es2021 or similar as deno doesn't have the same APIs as node.

Comment (Nemo157) [reply eslint.config.js:71]: Right, it's not using any globals so can just be removed from this override.

**Comment by GuillaumeGomez** [comment]
> I have an alternative using `node` with a `package.json` instead if anyone has opinions on which to use: [Nemo157@2f41680](https://github.com/Nemo157/docs.rs/commit/2f41680fd1e02274252ab0450d687e43cbea6dc8)

No preference on my side, so whichever sounds best to you.

**Comment by Nemo157** [comment]
My preference is this approach with deno, it avoids having to manage the whole `package-lock.json` when we don't really care about the transitive deps.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by syphar** [comment]
@Nemo157 @GuillaumeGomez I'm not sure I fully follow, is anything missing here? Or is this ready to be merged? 

Generally you two can merge too :)

**Comment by GuillaumeGomez** [comment]
I prefer to let PR owners with merge rights to merge their own PRs once approved so it's waiting on @Nemo157 here. ;)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Let's merge it then.

**Pull request merged**
